theano/sandbox/cuda/opt.py

```diff
@@ -2032,6 +2062,16 @@ def local_inplace_ger(node):
     if node.op == gpu_ger_no_inplace:
         return [gpu_ger_inplace(*node.inputs)]
 
+
+@local_optimizer([GpuCholesky(lower=True, inplace=False),
+                  GpuCholesky(lower=False, inplace=False)], inplace=True)
+def local_inplace_gpu_cholesky(node):
+    if node.op == GpuCholesky(lower=True, inplace=False):
```

Comment:
The full code of this method can be simplified:

if isinstance(node.op, GpuCholesky) and not node.op.inplace:
    return [GpuCholesky(node.op.lower, inplace=True)]

Constructing object just to compare is time consuming and would make optimization slower. So this code will be simpler and faster.